packages/workflow/src/ObservableObject.ts

```diff
@@ -39,14 +47,20 @@ export function create(target: IDataObject, parent?: IObservableObject, option?:
 		set(target, name, value) {
 			if (parent === undefined) {
 				// If no parent is given mark current data as changed
-				if (option !== undefined && option.ignoreEmptyOnFirstChild === true && depth === 0
-					&& target[name.toString()] === undefined && typeof value === 'object' && Object.keys(value).length === 0) {
+				if (
+					option !== undefined &&
+					option.ignoreEmptyOnFirstChild === true &&
+					depth === 0 &&
+					target[name.toString()] === undefined &&
+					typeof value === 'object' &&
+					Object.keys(value).length === 0
+				) {
 				} else {
-					(target as IObservableObject).__dataChanged = true;
+					(target as IObservableObject).__dataChanged = true; // eslint-disable-line no-underscore-dangle
 				}
 			} else {
 				// If parent is given mark the parent data as changed
-				parent.__dataChanged = true;
+				parent.__dataChanged = true; // eslint-disable-line no-underscore-dangle
```

Comment:
Underscores are used like this frequently throughout the codebase, so it would seem we will have many exceptions if we enforce this rule.

Reply:
it is not generally forbidding underscores, but certain patterns. Happy to discuss if we should kick out this rule. I kept it, because if it is used the Programmer can disable it and is doing it on purpose then. However I think for general coding it should be checked and forbidden.

reference: https://eslint.org/docs/rules/no-underscore-dangle